### DISCUSSION TITLE
Does jupyter-book work with Python 3.9 on windows?

### DISCUSSION THREAD

**Discussion by hsauro** [executablebooks/meta#422] [Q&A]
I see from the docs that jupyter-books does not work on windows when using Python 3.8. I assume that is the same for Python 3.9 and the upcoming 3.10? We're currently on Python 3.8, probably moving to 3.9 soon and it's a shame that jupyter-books only works on older versions of Python. Are there plans to get it working on more up-to-date version of Python?

**Comment by choldgraf** [comment]
Hmmm - the answer is that I'm not sure yet because it's not [yet tested on Python 3.9](https://github.com/executablebooks/jupyter-book/blob/master/.github/workflows/tests.yml#L86). 

What doesn't work when you use it with 3.8 on Windows?

**Comment by hsauro** [comment]
We're told on this page that it doesn't work on Windows Python 3.8

https://jupyterbook.org/advanced/windows.html#working-on-windows

so i assumed it didn't. We could give it a try to see if it does work. I
have it working with Python 3.7 but our main workhorse at the moment is
3.8.

Herbert

On Thu, Jul 8, 2021 at 12:42 PM Chris Holdgraf ***@***.***>
wrote:

> Hmmm - the answer is that I'm not sure yet because it's not yet tested on
> Python 3.9
> <https://github.com/executablebooks/jupyter-book/blob/master/.github/workflows/tests.yml#L86>
> .
>
> What doesn't work when you use it with 3.8 on Windows?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/executablebooks/meta/discussions/422#discussioncomment-981131>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIBSDWC4XR6PPQPIBU4WWTTWX5OPANCNFSM475EW5YQ>
> .
>


-- 
Herbert Sauro, Professor
University of Washington, Bioengineering
206-685-2119, www.sys-bio.org
***@***.***
Books: http://books.analogmachine.org/

**Reply by choldgraf** [reply]
You could certainly give it a shot in windows. It may be time to up the python version we test there

**Comment by hsauro** [comment]
ok we'll give it a go.

H

On Thu, Jul 8, 2021 at 1:53 PM Chris Holdgraf ***@***.***>
wrote:

> You could certainly give it a shot in windows. It may be time to up the
> python version we test there
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/executablebooks/meta/discussions/422#discussioncomment-981367>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAIBSDUKGLW5STCJP5WC223TWYF4TANCNFSM475EW5YQ>
> .
>


-- 
Herbert Sauro, Professor
University of Washington, Bioengineering
206-685-2119, www.sys-bio.org
***@***.***
Books: http://books.analogmachine.org/

**Comment by hsauro** [comment]
Looks like it won't work with Python 3.8, I get the following error when I
try to create one chapter from the book.

"python-3.8.6.amd64\lib\asyncio\events.py", line 501, in add_reader  raise
NotImplementedError

asyncio is mentioned to be an issue in the jupyterbook website for windows (
https://jupyterbook.org/advanced/windows.html#working-on-windows). It talks
about this workaround:

Work-around: Pin to Python 3.7.6. This environment_win.yml
<https://github.com/eoas-ubc/quantecon-mini-example/blob/windows/environment_win.yml>
file
does that, and also installs runjb to fix issue 1.

But I'm not sure what that means or if it means we can use Python 3.8. Two
other issues are also listed, but both appear fairly benign for our book.

For now we'll switch back to Python 3.7.

Herbert

On Thu, Jul 8, 2021 at 1:55 PM Herbert M Sauro ***@***.***> wrote:

> ok we'll give it a go.
>
> H
>
> On Thu, Jul 8, 2021 at 1:53 PM Chris Holdgraf ***@***.***>
> wrote:
>
>> You could certainly give it a shot in windows. It may be time to up the
>> python version we test there
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/executablebooks/meta/discussions/422#discussioncomment-981367>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAIBSDUKGLW5STCJP5WC223TWYF4TANCNFSM475EW5YQ>
>> .
>>
>
>
> --
> Herbert Sauro, Professor
> University of Washington, Bioengineering
> 206-685-2119, www.sys-bio.org
> ***@***.***
> Books: http://books.analogmachine.org/
>


-- 
Herbert Sauro, Professor
University of Washington, Bioengineering
206-685-2119, www.sys-bio.org
***@***.***
Books: http://books.analogmachine.org/
